packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.base.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import {
+  autobind,
+  customizable
+} from '../../Utilities';
+import {
+  IColorCellProps,
+  IColorPickerGridCellProps,
+  IColorPickerGridCellStyleProps,
+  IColorPickerGridCellStyles
+} from './ColorPickerGridCell.types';
+import { getColorFromString } from '../../utilities/color/colors';
+import { GridCell } from '../../utilities/grid/GridCell';
+import { IGridCellProps } from '../../utilities/grid/GridCell.types';
+import { classNamesFunction, IClassNames } from '../../Styling';
+
+const getClassNames = classNamesFunction<IColorPickerGridCellStyleProps, IColorPickerGridCellStyles>();
+
+class ColorCell extends GridCell<IColorCellProps, IGridCellProps<IColorCellProps>> {
+}
+
+@customizable('ColorPickerGridCell', ['theme'])
+export class ColorPickerGridCellBase extends React.Component<IColorPickerGridCellProps, {}> {
+
+  public static defaultProps = {
+    circle: true,
+    disabled: false,
+    selected: false,
+  } as IColorPickerGridCellProps;
+
+  private _classNames: {[key in keyof IColorPickerGridCellStyles]: string };
+
+  public render() {
+    let {
+      item,
+      id,
+      selected,
+      disabled,
+      getStyles,
+      theme,
+      circle
+    } = this.props;
+
+    this._classNames = getClassNames(
+      getStyles!,
```

Comment:
Is the ! necessary here? Should getStyles be a required prop?

Reply:
I don't know how to make it "required" by the base, but not by the styled version.